sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,555 @@
+        -------
+        y_preds : ndarray, shape (n_samples, n_estimators)
+            Prediction outputs for each estimator.
+        """
+        check_is_fitted(self, 'estimators_')
+        return self._concatenate_predictions(
+            X, [getattr(est, meth)(X)
+                for est, meth in zip(self.estimators_, self.predict_method_)
+                if est not in (None, 'drop')]
+        )
+
+    @if_delegate_has_method(delegate='final_estimator_')
+    def predict(self, X, **predict_params):
+        """Predict target for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        **predict_params : dict of string -> object
+            Parameters to the ``predict`` called by the ``final_estimator``.
+            Note that while this may be
+            used to return uncertainties from some models with ``return_std``
+            or ``return_cov``.
+
+        Returns
+        -------
+        y_pred : ndarray, shape (n_samples,) or (n_samples, n_output)
+            Predicted targets.
+        """
+
+        check_is_fitted(self, ['estimators_', 'final_estimator_'])
+        return self.final_estimator_.predict(self.transform(X))
+
+
+class StackingClassifier(_BaseStacking, ClassifierMixin):
+    """Stacked of estimators using a final classifier.
+
+    Stacked generalization consists in stacking the output of individual
+    estimator and use a classifier to compute the final prediction. Stacking
+    allows to combine the strength of each individual estimator. It should be
+    noted that the final estimator is trained through cross-validation.
+
+    .. versionadded:: 0.20
+
+    Read more in the :ref:`User Guide <stacking>`.
+
+    Parameters
+    ----------
+    estimators : list of (string, estimator) tuples
+        Base estimators which will be stacked together. An estimator can be set
+        to None or 'drop' using ``set_params``.
+
+    final_estimator : estimator object
+        A classifier which will be used to combine the base estimators.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy. Possible inputs for
+        cv are:
+
+        * None, to use the default 3-fold cross validation,
+        * integer, to specify the number of folds in a (Stratified) KFold,
+        * An object to be used as a cross-validation generator,
+        * An iterable yielding train, test splits.
+
+        For integer/None inputs, if the estimator is a classifier and y is
+        either binary or multiclass, StratifiedKFold is used. In all other
+        cases, KFold is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    predict_method : list of string or 'auto', optional
```

Comment:
Does it make sense to use `predict` with classifiers, considering that none of our estimators can handle categorical data? Should we add a note about this? Classes will be treated as ordered values which doesn't seem appropriate.

Reply:
You can still pass a pipeline as a final estimator with a one-hot encoding.

Reply:
you can, but the point is that you actually **should**.

And it's not obvious to know which columns should be treated as categorical. More generally I formalized my concerns in https://github.com/scikit-learn/scikit-learn/pull/11047#issuecomment-502230627